src/frontend/src/icp/components/core/IcLoaderWallets.svelte

```diff
@@ -0,0 +1,71 @@
+<script lang="ts">
+	import { onDestroy } from 'svelte';
+	import { initIcrcWalletWorker } from '$icp/services/worker.icrc-wallet.services';
+	import { initIcpWalletWorker } from '$icp/services/worker.icp-wallet.services';
+	import type { WalletWorker } from '$icp/types/ic-listener';
+	import { enabledIcNetworkTokens } from '$lib/derived/network-tokens.derived';
+	import type { TokenId } from '$lib/types/token';
+	import { debounce, nonNullish } from '@dfinity/utils';
+	import { toastsError } from '$lib/stores/toasts.store';
+	import { i18n } from '$lib/stores/i18n.store';
+	import { replacePlaceholders } from '$lib/utils/i18n.utils';
+
+	let workers: Map<TokenId, WalletWorker> = new Map<TokenId, WalletWorker>();
+
+	const manageWorkers = async () => {
+		const unusedWorkers: TokenId[] = Array.from(workers.keys()).filter(
```

Comment:
maybe extract this step, cleaning up worker for tokens that are not active anymore, in a dedicated function? For readability and maintanability reasons

Reply:
done, created util function